src/App.tsx

```diff
@@ -3,32 +1,100 @@
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { Completed, Filters } from './types/Filters';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filters, setFilters] = useState<Filters>({
+    completedType: Completed.All,
+    searchByText: '',
+  });
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  const selectTodo = (todo: Todo | null) => {
+    setSelectedTodo(todo);
+  };
+
+  const updateFilters = (key: keyof Filters, value: string | Completed) => {
+    setFilters(prevFilters => ({
+      ...prevFilters,
+      [key]: value,
+    }));
+  };
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter(item => {
+      const matchesCompletedType =
+        filters.completedType === Completed.All ||
+        (filters.completedType === Completed.Completed && item.completed) ||
+        (filters.completedType === Completed.Active && !item.completed);
+
+      const matchesSearchText = item.title
+        .toLowerCase()
+        .includes(filters.searchByText.toLowerCase());
+
+      return matchesCompletedType && matchesSearchText;
+    });
+  }, [todos, filters]);
+
+  useEffect(() => {
+    getTodos()
+      .then(todosData => {
+        setTodos(todosData);
+      })
+      .catch(error => {
+        setErrorMessage(`Failed to fetch todos: ${error.message}`);
```

Comment:
Ensure that `error.message` is a valid property. If the error object structure is different, this might lead to undefined messages. Consider logging the error to verify its structure.